test/concurrent/ThreadTest.ooc

```diff
@@ -45,15 +45,17 @@ ThreadTest: class extends Fixture {
 				value set(value get() + 1)
 				Thread yield()
 				Time sleepMilli(1)
+				if (Thread currentThread() isCanceled())
```

Comment:
> Do you find my idea as ugly as I do

Yes. :) 

I have an idea about creating a `ThreadData` class which could contain the mutex and the flags and all the associated logic, and then `currentThread()` would just pass that instance along to the new thread instance. No fuss and the thread class itself is kept clean. 

I'm not saying your idea isn't good, but I'm not completely sold as it's more complex to implement and the java approach just isn't as great without interfaces (`Runnable`).